rs/execution_environment/src/scheduler/scheduler_metrics.rs

```diff
@@ -675,6 +677,15 @@ impl SchedulerMetrics {
                 "scheduler_stop_canister_calls_without_call_id",
                 "Number of stop canister calls with missing call ID.",
             ),
+            canister_snapshots_memory_usage: memory_histogram(
+                "scheduler_canister_snapshots_memory_usage_bytes",
+                "Canisters total snapshots memory usage distribution in bytes.",
+                metrics_registry,
+            ),
```

Comment:
I think this shouldn't be a histogram, because it's a single subnet-wide number which gets updated each round. So it should really be an `IntGauge` as well, right? I think the histogram would be what we used if we were tracking this per canister, but if I'm understanding correctly this is just the total usage.

Reply:
Good point, I think technically we could also report per canister and keep it as a histogram but it would come at some cost because we we would need to compute the per canister memory usage every time. We could also keep track of it but I'm not sure it's worth it. I'd keep the total only for now and convert to gauge unless we believe the per canister usage would be very interesting.

Reply:
@berestovskyy @schneiderstefan Do you guys have an opinion here? Would a histogram by canister snapshot memory usage would be more useful or is the total good enough for now?

Reply:
I don't have a strong opinion. A breakdown per canister would of course be an extra signal, but I'm not sure if it's a particularly useful one.

Reply:
The description says "snapshots memory **distribution**", so it makes sense to have it as a histogram. I guess Adam is pointing to the fact that we observe not a per-canister value, but the `final_state.canister_snapshots.memory_taken()`.

Reply:
Yes I was just saying we should either track the per canister usage as a histogram, or the subnet total usage as a single integer. But I have no opinion on which of the two we use.

Reply:
> The description says "snapshots memory distribution", so it makes sense to have it as a histogram

That's because I copied over from other histograms. In order to observe per canister values and have acceptable performance, we'll need some more changes to keep track of per canister memory usage so we don't need to recompute it on every round. I was mostly interested to hear if people think it's worth it and if yes, I can make the changes to report per canister usage, otherwise we stick to single value per subnet (and make it a gauge).

Reply:
A gauge SGTM.